### PR TITLE
Add examples for NXP JN516x using TSCH

### PR DESCRIPTION
**Pull request by TeVeDe**
Examples for the JN516x using TSCH have been transferred to the main line.
Make file for the regression tests is updated as well.

### CODE REVIEW

File: examples/jn516x/tsch/tools/rich.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+rich_init(uip_ipaddr_t *br_prefix)
+{
+#if TSCH_CONFIG == TSCH_CONFIG_ORCHESTRA
+  orchestra_init();
+#endif
+
+  rich_network_init(br_prefix);
+
+#if WITH_COAP_RESOURCES
+  plexi_init();
```
Comment (simonduq) [inline examples/jn516x/tsch/tools/rich.c]: undefined ref to plexy, should be removed

File: examples/jn516x/tsch/common-conf.h [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+#undef ENABLE_COOJA_DEBUG
+#define ENABLE_COOJA_DEBUG 0
+
+#if WITH_COAP_RESOURCES
+#define UIP_DS6_WITH_LINK_METRICS 1
+#define PLEXI_CONFIG_RPL 1
+#if WITH_TSCH
+#define PLEXI_CONFIG_TSCH            2
+#endif /* WITH_TSCH */
+#define RICH_INTERFACE PLEXI_CONFIG_RPL | PLEXI_CONFIG_TSCH
```
Comment (simonduq) [inline examples/jn516x/tsch/common-conf.h]: above are a few plexi-related defines, should be removed as plexi is not in upstream contiki

File: examples/jn516x/tsch/tx-power-verification/node/Makefile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+TARGET ?= jn516x
+JN516x_WITH_DONGLE = 1
+
+CONTIKI=../../../../..
+CONTIKI_WITH_IPV6 = 1
+
+CFLAGS += -DPROJECT_CONF_H=\"project-conf.h\"
+PROJECTDIRS += .. ../../tools
+PROJECT_SOURCEFILES += rich.c
```
Comment (simonduq) [inline examples/jn516x/tsch/tx-power-verification/node/Makefile]: I would suggest renaming rich (no meaning for contiki upstream) to rpl-tools, as it was done for https://github.com/contiki-os/contiki/tree/master/examples/jn516x/rpl/tools

(will also improve consistency accros examples)

File: examples/jn516x/tsch/uart1-test-node/uart1-test-node.c [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    tail_index = ((tail_index + 1)& RINGBUF_MAX_INDEX);
+    *c = ringbuf[tail_index];
+  } else {
+    return_val = -1;
+  }
+  uart1_enable_interrupts();
+  return return_val;
+}
+
+static int put_ringbuf(uint8_t c)
```
Comment (simonduq) [inline examples/jn516x/tsch/uart1-test-node/uart1-test-node.c]: code style: `static int` should be alone on a first line. I saw a few more instances above

**Comment by simonduq** [comment]
Thanks, ready to merge after the comments are addressed

**Comment by TeVeDe** [comment]
Feedback processed in added examples

**Comment by simonduq** [comment]
Looks good to me :+1:

**Pull request merged**